certora/specs/RevertsMintBurnEthereum.spec

```
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+import "Delegation.spec";
+
+methods {
+    function owner() external returns address envfree;
+    function totalSupply() external returns uint256 envfree;
+    function balanceOf(address) external returns uint256 envfree;
+    function delegatee(address) external returns address envfree;
+    function delegatedVotingPower(address) external returns uint256 envfree;
+}
+
+// Check the revert conditions for the burn function.
+rule mintRevertConditions(env e, address to, uint256 amount) {
+    requireInvariant zeroAddressNoVotingPower();
+    assert isTotalSupplyGTEqSumOfVotingPower();
+    mathint totalSupplyBefore = totalSupply();
+    uint256 balanceOfSenderBefore = balanceOf(e.msg.sender);
+    uint256 toVotingPowerBefore = delegatedVotingPower(delegatee(to));
+
+    // Safe require because if the delegatee is not zero the recipient's delegatee's voting power is lesser than or equal to the total supply of tokens.
+    require delegatee(to) != 0 => toVotingPowerBefore <= totalSupply();
+
+    // Safe require thats avoid absurd counter-examples introduced by munging the sources.
+    require toVotingPowerBefore <= sumOfVotingPower;
```

```suggestion
    // Safe require because of the invariant XXX.
    require toVotingPowerBefore <= sumOfVotingPower;
```
this is an invariant that we could prove using the same pattern as for the total supply, where the property XXX is "`sumOfVotingPower` is greater than any one voting power"